Snake/Snake/Reader.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Threading;
+
+namespace Snake
+{
+    // This class is used as a key listner. It waits for a few milliseconds for a key to be pressed.
+    // Advantages of using this class: No busy waiting.
+
+    // Modified version of: http://stackoverflow.com/questions/57615/how-to-add-a-timeout-to-console-readline
+    class Reader
+    {
+        private static AutoResetEvent getInput, gotInput;
+        private static ConsoleKeyInfo keyInfo = new ConsoleKeyInfo();
+
+        static Reader()
+        {
+            getInput = new AutoResetEvent(false);
+            gotInput = new AutoResetEvent(false);
+            Thread inputThread = new Thread(reader);

```

Comment:
so you are using threads. this is probably necessary but you have a truck load of statics in this application so it thread safe? we can talk about this further if you like but you need to think about all the things that can go wrong when you use statics and multiple threads